api/reviews.go

```diff
@@ -134,3 +137,61 @@ func (api *API) getReviewHandler(writer http.ResponseWriter, request *http.Reque
 
 	log.Event(ctx, "successfully retrieved review", log.INFO, logData)
 }
+
+func (api *API) updateReviewHandler(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	bookID := mux.Vars(request)["id"]
+	reviewID := mux.Vars(request)["reviewID"]
+
+	logData := log.Data{"book_id": bookID, "review_id": reviewID}
+
+	if bookID == "" {
+		handleError(ctx, writer, apierrors.ErrEmptyBookID, logData)
+		return
+	}
+
+	if reviewID == "" {
+		handleError(ctx, writer, apierrors.ErrEmptyReviewID, logData)
+		return
+	}
+
+	// Confirm that book exists. If bookID not found, then do not check for the review
+	_, err := api.dataStore.GetBook(ctx, bookID)
```

Comment:
Datastore could return an error for multiple reasons so I'd say it's probably best not to assume that error ==  book not found. A common pattern for Go libraries it to return a public error var specifically for not found. If error != nil the caller can check  if err == TheNotFound error and act accordingly. Maybe that is something you could consider here.

Reply:
I've now made sure that the `datastore` is returning an error without creating a cyclic dependency. 

Here, I am using the `handleError` function to do the equivalent of the common pattern you described. Does this make sense?